src/app/api/api_v1/endpoints/detections.py

```diff
@@ -64,23 +64,40 @@ async def create_detection(
 @router.get("/{detection_id}", status_code=status.HTTP_200_OK, summary="Fetch the information of a specific detection")
 async def get_detection(
     detection_id: int = Path(..., gt=0),
+    cameras: CameraCRUD = Depends(get_camera_crud),
     detections: DetectionCRUD = Depends(get_detection_crud),
     token_payload: TokenPayload = Security(get_jwt, scopes=[UserRole.ADMIN, UserRole.AGENT, UserRole.USER]),
 ) -> Detection:
     telemetry_client.capture(token_payload.sub, event="detections-get", properties={"detection_id": detection_id})
-    return cast(Detection, await detections.get(detection_id, strict=True))
+    detection = cast(Detection, await detections.get(detection_id, strict=True))
+    camera = await cameras.get(detection.camera_id, strict=True)
+    if (
+        camera is not None
+        and token_payload.organization_id != camera.organization_id
+        and UserRole.ADMIN not in token_payload.scopes
+    ):
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="Access forbidden.")
+    return detection
```

Comment:
more of a product discussion than backend: I agree that org should prevent cross access on users, and cameras. On detection, I don't know :thinking: 

If we want to do it:
- let's add a `if UserTole.ADMIN in token_payload.scopes` and return the previous line
- the other verif (otherwise we fetch a camera for nothing when this is an admin, considering the camera_id verification will happen during the DB operation either way)